control-plane/pkg/reconciler/channel/v2/channelv2.go

```diff
@@ -576,32 +583,47 @@ func (r Reconciler) reconcileConsumerGroup(ctx context.Context, channel *messagi
 		},
 	}
 
+	// TODO: make keda annotation values configurable and maybe unexposed
+	expectedCg.Annotations = kedafunc.SetAutoscalingAnnotations(channel.Annotations)
```

Comment:
should we propagate subscription annotations more than channel ones to have more granularity and consistently like triggers?

Reply:
Hmm is there an advantage to that granularity? Do we expect subscribers for the same channel to have different KEDA settings?

Reply:
Yes, because for example different consumer want to move at different pace based on how fast they can process events, so they might want to set min and max replicas differently

Reply:
Ok. Though a channel type does not have the full `Subscribable` resource though (if its even called a resource) just the addressable info from the Spec. I assume it can be of any K8 type so getting the actual object to find its annotations might not be something we can do?

Reply:
We can, and that's what I'm doing for https://github.com/knative-sandbox/eventing-kafka-broker/pull/2651

Reply:
I think we can extract a common function

Reply:
My example KafkaChannel (that works) looks like this:
```
apiVersion: messaging.knative.dev/v1beta1
kind: KafkaChannel
metadata:
  name: kafka-channel
  annotations:
    autoscaling.knative.dev/class: keda.autoscaling.knative.dev
    autoscaling.knative.dev/minScale: "0" 
    autoscaling.knative.dev/maxScale: "5" 
    keda.autoscaling.knative.dev/pollingInterval: "3" 
    keda.autoscaling.knative.dev/cooldownPeriod: "10" 
    keda.autoscaling.knative.dev/kafkaLagThreshold: "2"  
spec:
  numPartitions: 10
  replicationFactor: 3
  
  :
    - subscriberUri: http://eventdisplay.default.svc.cluster.local/
      uid: "5e4e9b73-927d-4a5a-a7b3-1f3c8d9ac0da"
      
kind: Service
apiVersion: v1
metadata:
  name: eventdisplay
  annotations:
      prometheus.io/scrape: 'true'
      prometheus.io/port:   '8001'
spec:
  selector:
    app: eventdisplay
  ports:
    - protocol: TCP
      port: 80
      targetPort: 8080
      name: http
```
where `subscribers.ui` is the UID of the K8 service, and not a `Subscription` object

```
 ansuvarghese$ k get subscriptions --all-namespaces
No resources found
```

Reply:
@pierDipi wdyt about this kinda channel installs?

Reply:
I'd say that when a subscriber is created without a subscription it cannot be scaled with KEDA

I don't even know if that should be even valid.